tests/unit/cli/manager/test_config_manager.py

```diff
@@ -477,6 +472,51 @@ def test__get_train_type(self, mocker):
         config_manager.template.hyper_parameters.parameter_overrides = {}
         assert config_manager._get_train_type(ignore_args=True) == "Incremental"
 
+        config_manager.args.train_type = None
+        config_manager.args.unlabeled_data_roots = None
+        config_manager.args.train_data_roots = "tests/assets/classification_dataset"
+        config_manager.args.val_data_roots = "tests/assets/classification_dataset"
+        assert config_manager._get_train_type(ignore_args=False) == "Incremental"
+        config_manager.args.unlabeled_data_roots = "tests/assets/unlabeled_dataset/a"
+        assert config_manager._get_train_type(ignore_args=False) == "Semisupervised"
+        # test unlabeled_data_roots doesn't exist
+        config_manager.args.unlabeled_data_roots = "non_exist_dir"
+        try:
+            config_manager._get_train_type(ignore_args=False)
+            assert False
+        except ValueError:
+            assert True
+        config_manager.args.train_data_roots = "non_exist_dir"
+        config_manager.args.val_data_roots = "tests/assets/classification_dataset"
+        config_manager.args.unlabeled_data_roots = None
+        try:
+            config_manager._get_train_type(ignore_args=False)
+            assert False
+        except ValueError:
+            assert True
+        config_manager.args.train_data_roots = "tests/assets/unlabeled_dataset/a"
+        config_manager.args.val_data_roots = "tests/assets/unlabeled_dataset/a"
+        # test folder with only images
+        assert config_manager._get_train_type(ignore_args=False) == "Selfsupervised"
+        config_manager.args.val_data_roots = None
+        config_manager.args.train_data_roots = "tests/assets/classification_dataset"
+        # test val_data_roots is None -> Incremental (auto_split)
+        assert config_manager._get_train_type(ignore_args=False) == "Incremental"
+        try:
+            os.mkdir("tests/assets/classification_dataset/unlabeled_images")
+            config_manager.args.train_data_roots = "tests/assets/classification_dataset"
+            config_manager.args.val_data_roots = "tests/assets/classification_dataset"
+            # test empty unlabeled folder
+            assert config_manager._get_train_type(ignore_args=False) == "Incremental"
+            os.mkdir("tests/assets/classification_dataset/unlabeled_images/some")
+            # test unsufficient number of images
+            assert config_manager._get_train_type(ignore_args=False) == "Incremental"
+        finally:
+            shutil.rmtree("tests/assets/classification_dataset/unlabeled_images")
```

Comment:
If you want to use temporary directory removed automatically, you can use `tempfile.TemporaryDirectory`.

Reply:
I used os.mkdir because I need folder with exact name and exact location. In tempfile I didn't find such functionality. Do you know other way how to do that?